__tests__/app.test.js

```diff
@@ -0,0 +1,125 @@
+const mongoose = require('mongoose');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const request = require('supertest');
+const app = require('../lib/app');
+const Bird = require('../lib/models/Bird');
+
+describe('app routes', () => {
+  const mongo = new MongoMemoryServer();
+  beforeAll(() => {
+    return mongo.getUri()
+      .then(uri => mongoose.connect(uri, {
+        useUnifiedTopology: true,
+        useNewUrlParser: true
+      }));
+  });
+
+  beforeEach(() =>{
+    return mongoose.connection.dropDatabase();
+  });
+
+  afterAll(() => {
+    return mongoose.connection.close()
+      .then(()=>mongo.stop());
+  });
+
+  it('creates a new bird', () => {
+    return request(app)
+      .post('/birds')
+      .send({ bird: 'barn owl', season: 'summer' })
+      .then(res => {
+        expect(res.body).toEqual({
+          _id: expect.anything(),
+          bird: 'barn owl',
+          season: 'summer',
+          spotted: 1,
+          __v: 0
+        });
+      });
+  });
+
+  it('gets a list of birds', async() => {
+    await Bird.create({
+      bird: 'sparrow',
+      season: 'fall'
+    });
+
+    return request(app)
+      .get('/birds')
+      .then(res => {
+        expect(res.body).toEqual([
+          {
+            _id: expect.anything(),
+            bird: 'sparrow',
+            season: 'fall',
+            spotted: 1,
+            __v: 0
+          }
+        ]);
+      });
+  });
+
+  it('gets a bird by id', async() => {
+    const { _id } = await Bird.create({
+      bird: 'raven',
+      season: 'winter'
+    });
+
+    return request(app)
+      .get(`/birds/${_id}`)
+      .then(res => {
+        expect(res.body).toEqual({
+          _id: _id.toString(),
+          bird: 'raven',
+          season: 'winter',
+          spotted: 1,
+          __v: 0
+        });
+      });
+  });
+
+  it('deletes a bird by id', async() => {
```

Comment:
move this test under the update test. Keep it in CRUD order. Helps with readability. Tired eyes like to see stuff in an order that they expect.